### ISSUE TITLE
Some training issues

### ISSUE THREAD

**Issue by Linql7** [multimediaFor/ProFact#2]
Hello author, I encountered some challenges while trying to use this project and am unsure how to solve them.

1：Regarding the second training stage, is it continuing to fine tune on the "best model" saved in the first stage? If so, is the corresponding code in the `config. py` as follows:
    `def get_option():
    parser = argparse.ArgumentParser(description="Training Parameter Setting")
    # model
    parser.add_argument("--ver", type=str, default="b3", help="base model")
    parser.add_argument("--pretrained", type=str, default=True, help="use or not use pretrained weight")   
    parser.add_argument("--loaded", type=str, default=False, help="load weight")   
    parser.add_argument("--load_path", type=str, default="", help="load weight")   `
2：Regarding training time, what is the approximate time for you to complete 50 epochs of training using a multi card or single card A800 (40G) or A800 (80G)?

If you have the time and opportunity to provide some guidance or advice, I would greatly appreciate it. Thank you for contributing such a valuable project to the community!

**Comment by multimediaFor** [comment]
Thank you for your interest in our work.
1. Yes, the second training stage is to continue fine-tuning on the "best model" saved in the first stage. The corresponding parameters in the `config.py` are set as follows:
 `def get_option():   
    parser = argparse.ArgumentParser(description="Training Parameter Setting")  
    parser.add_argument("--ver", type=str, default="b3", help="base model")  
    parser.add_argument("--pretrained", type=str, default=False, help="use or not use pretrained weight")  
    parser.add_argument("--loaded", type=str, default=True, help="load weight")  
    parser.add_argument("--load_path", type=str, default="./checkpoint_save/best_model.pth", help="load weight")
`
2. Using a single A800 (80G) card, it takes approximately 45 hours to complete 50 epochs of training in the first stage.

**Comment by Linql7** [comment]
Thank you very much for your reply.
I have encountered a new training bug and have found a possible solution. I hope you can help me check if it is correct.

I don't know if you have noticed that during the training process of this code, the server's memory (not GPU memory) will continue to increase significantly. Especially when using large-scale datasets, it is more evident.

I noticed between lines 170 and 189 in the `train.py` you provided.

```
now_train_loss += loss * batch_size
            now_iter_size  += batch_size
            # update tdqm bar message
            train_tbar.set_postfix(loss = (now_train_loss/now_iter_size).item())

            f1, iou = get_metrics(dets, gts)
            now_train_f1 += f1 * batch_size
            now_train_iou += iou * batch_size

            if i % 30 == 0:
                iters = i + (epoch - 1) * len(train_loader)
                writer.add_scalar('train/loss', (now_train_loss/now_iter_size).item(), iters)
        
        # update learing rate scheduler
        scheduler.step()  

        logger.write('Epoch: {}, Train Loss: {:.6f}, F1: {:.5f}, mIoU: {:.5f}'.format(epoch, 
                                                                               (now_train_loss/now_iter_size).item(), 
                                                                               (now_train_f1/now_iter_size), 
                                                                               (now_train_iou/now_iter_size)))
```

The main cause of this issue is `now_train_ross+=loss * batch_stize` .
I think this line of code should be written as `now_train-loss+=loss. item() * batch_stize`
Correspondingly,
The 173rd line should be changed to: `train_tbar.set_postfix (loss=(now_train_Loss/now_iter_Size))`
The 181st line should be changed to: `writer. add_stcalar ('train/miss', (now_train_rass/now_iter_2). item(), iters)`
The 187th line should be changed to: `(now_train_Loss/now_iter_Size),`


After the changes, the server's memory is no longer gradually increasing, but tends to stabilize.
But I'm not sure if this change will affect the model's performance. I would like to confirm and ask for your advice

**Comment by multimediaFor** [comment]
Thank you for bringing this memory issue to my attention. By converting the loss Tensor into a Python numeric value using .item(), it typically helps reduce memory consumption by avoiding the potential memory overhead associated with storing large Tensor objects. Theoretically, this change should not impact the model's training performance, as it merely alters the manner in which the loss and evaluation metrics are computed without altering the model's learning process or gradient updates.

**Comment by Linql7** [comment]
Thank you very much for your reply
